File: tests/function/dsp_def/function_dsp_def.ml
open Optitrust
open Target


let _ = Run.script_cpp (fun _ ->

  !! Function_basic.dsp_def [cFunDef "test_simpl"];
  !! Function_basic.dsp_def [cFunDef "test_one_branch"];
  !! Function_basic.dsp_def [cFunDef "test_branches"];

  !! Function_basic.dsp_def ~func:"my_test_simpl" ~arg:"my_res" [cFunDef "test_simpl"];
)

File: tests/function/dsp_def/function_dsp_def.cpp
int test_simpl(int x){
    
    return x;
}

// FIXME: should add early return
int test_one_branch(int x){
    if (x < 0){
      return -x;
    }
    return x;
}

int test_branches(int x){
  if (x > 0){
    return x;
  }
  else {
    return -x;
  }
}

int main(){
  return 0;
}